src/scripts/main.js

```diff
@@ -1,3 +1,279 @@
 'use strict';
 
-// write your code here
+const rowsNum = 4;
+const columnsNum = 4;
+
+const field = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+const startButton = document.querySelector('.button');
+const gameField = document.querySelector('.game-field').tBodies[0];
+
+const gameScore = document.querySelector('.game-score');
+const messageLose = document.querySelector('.message-lose');
+const messageWin = document.querySelector('.message-win');
+const messageStart = document.querySelector('.message-start');
+
+const winValue = 2048;
+let maxTileValue = 0;
+let score = 0;
+let isListenerRemoved = false;
+
+function filterZero(row) {
+  return row.filter(tileValue => tileValue !== 0);
+}
+
+function slide(row, isChecking = false) {
+  let newRow = filterZero(row);
+
+  for (let i = 0; i < newRow.length - 1; i++) {
+    if (newRow[i] === newRow[i + 1]) {
+      newRow[i] *= 2;
+      newRow[i + 1] = 0;
+
+      if (!isChecking) {
+        score += newRow[i];
+
+        maxTileValue = newRow[i] > maxTileValue
+          ? newRow[i]
+          : maxTileValue;
+      }
+    }
+  }
+
+  newRow = filterZero(newRow);
+
+  while (newRow.length < columnsNum) {
+    newRow.push(0);
+  }
+
+  return newRow;
+}
+
+function slideUp() {
+  for (let c = 0; c < columnsNum; c++) {
+    const row = slide([
+      field[0][c],
+      field[1][c],
+      field[2][c],
+      field[3][c],
```

Comment:
one small thing that I noticed, it's not a good idea to hardcode these indexes, what if we have a field with 5x5? 🙂